ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/locator/AllHostsLocator.java

```diff
@@ -51,6 +52,7 @@ public AllHostsLocator(final Supplier<List<PersistentQueryMetadata>> allPersiste
     }
 
     return currentQueries.stream()
+        .filter(persistentQueryMetadata -> persistentQueryMetadata.getState() == State.RUNNING)
```

Comment:
This only is used at the moment at the start of the query to know which nodes we should contact.  The idea is that at the moment, if the query isn't running anywhere yet, we should get an error.  This works in RQTT at the moment so that we can retry many times until everything is up and running.  The alternative is that we can block and poll the state until it's running, but that seems less ideal.  We might want to also try to fail until it's running everywhere but that's beyond what this change does.